pyccel/codegen/printing/fcode.py

```diff
@@ -408,10 +408,18 @@ def _handle_inline_func_call(self, expr, assign_lhs = None):
         self._additional_imports.update(func.imports)
         if func.global_vars or func.global_funcs:
             mod = func.get_direct_user_nodes(lambda x: isinstance(x, Module))[0]
-            self._additional_imports.add(Import(mod.name, [AsName(v, v.name) \
-                for v in (*func.global_vars, *func.global_funcs)]))
-            for v in (*func.global_vars, *func.global_funcs):
-                self.scope.insert_symbol(v.name)
+            sc  = self.scope
+            imp = True
+            while sc.parent_scope is not None:
+                if mod.scope is sc.parent_scope:
+                    imp = False
+                sc = sc.parent_scope
+
+            if imp:
+                self._additional_imports.add(Import(mod.name, [AsName(v, v.name) \
+                    for v in (*func.global_vars, *func.global_funcs)]))
+                for v in (*func.global_vars, *func.global_funcs):
+                    self.scope.insert_symbol(v.name)
```

Comment:
yes correct, the problem appears in Fortran only

Reply:
Ok. Can we use the shorter version. I find 8 lines easier to read than 17

Reply:
For some reason, your suggested solution does not work because the line
`current_mod = expr.get_direct_user_nodes(lambda x: isinstance(x, Module))[0]`
raises an error because the list is empty

Reply:
Maybe because of the use of `direct`. This would return an empty list for functions in functions. The expression should rather be:
`current_mod = expr.get_user_nodes(Module, excluded_nodes=(FunctionCall,))[0]`

Reply:
done in [bd593e5](https://github.com/pyccel/pyccel/pull/1728/commits/bd593e50538c77e3943b955921c0690d0866d708)